### DISCUSSION TITLE
libvirt: Vagrant box arista/veos  - The machine is in the 'inaccessible' state.

### DISCUSSION THREAD

**Discussion by sajith-kumar** [ipspace/netlab#1280] [General]
**I am Using libvirt/KVM with Vagrant**
I used netlab install command to install all the necessary packages.

**cat topology.yml**
---
defaults.device: eos
module: [ ospf ]

nodes: [ e1, c1, c2, e2 ]

links:
- e1-c1
- e1-c2
- c1:
  e2:
  bandwidth: 1000
- c2:
  e2:
  bandwidth: 1000

**When I run netlab up command or vagrant up command it gives below errors:**

==> c1:  -- Graphics Password:  Not defined
==> c2: Domain launching with graphics connection settings...
==> c1:  -- Graphics Websocket: 5701
==> c2:  -- Graphics Port:      5902
==> c2:  -- Graphics IP:        127.0.0.1
==> c2:  -- Graphics Password:  Not defined
==> c1: Waiting for domain to get an IP address...
==> e1: Creating shared folders metadata...
==> c2:  -- Graphics Websocket: 5702
==> e1: Starting domain.
==> c2: Waiting for domain to get an IP address...
==> e1: Domain launching with graphics connection settings...
==> e1:  -- Graphics Port:      5903
==> e1:  -- Graphics IP:        127.0.0.1
==> e1:  -- Graphics Password:  Not defined
==> e1:  -- Graphics Websocket: 5703
==> e1: Waiting for domain to get an IP address...
==> c2: Waiting for machine to boot. This may take a few minutes...
    c2: SSH address: 192.168.121.103:22
    c2: SSH username: vagrant
    c2: SSH auth method: private key
==> e2: Waiting for machine to boot. This may take a few minutes...
    e2: SSH address: 192.168.121.104:22
    e2: SSH username: vagrant
    e2: SSH auth method: private key
,,,,,,,,,,

later part of the error

An error occurred while executing the action on the 'c1'
machine. Please handle this error then try again:

The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'running'. The machine is in the
'inaccessible' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

The primary issue for this error is that the provider you're using
is not properly configured. This is very rarely a Vagrant issue
.........
Error executing vagrant up --provider libvirt:
  Command '['vagrant', 'up', '--provider', 'libvirt']' returned non-zero exit status 1.
[FATAL]   netlab up: vagrant up --provider libvirt failed, aborting.

**EOS - zerotouch** 
localhost#conf t
localhost(config)#show zerotouch

ZeroTouch Mode: Disabled
Script Exec Timeout: 900
localhost(config)#
localhost(config)#

**vagrant box list**
arista/veos (libvirt, 1)

Please help me figure out what I'm doing wrong.

**Comment by ipspace** [comment]
A number of questions:

* What version of EOS are you using?
* How did you build the Arista box? Did you use **netlab libvirt package** process?
* What is your environment? Are you using a virtual machine or a bare-metal server?
* What version of Ubuntu are you using?

You can connect to the Arista box while it's booting:

* Do "virsh list" to see VM names
* Do "virsh console _vm_" to connect to the VM

After that, can you figure out why the VM is inaccessible?

* Is the device configuration similar to what "netlab libvirt config eos" would give you?
* What's the state of the management interface?
* Does it have an IP address?
* Is the SSH server running?

Finally, would you be OK with Arista containers? ;)

**Comment by sajith-kumar** [comment]
The issue with my Vagrant libvirt installation has been resolved by following the instructions in this link(https://gist.github.com/PaulNeumann/81f299a7980f0b74cec9c5cc0508172b). 
Both Libvirt and Containerlabs are now working.

**Reply by ipspace** [reply]
Interesting, the only differences I could spot is that (A) the gist added user to 'kvm' group and (B) it did not pin the vagrant-libvirt version. Have to check the latest 'vagrant-libvirt' plugin with Ubuntu 22.04 and 24.04.

Thanks for the feedback! Ivan
